pkg/scheduler/objects/application.go

```diff
@@ -1847,7 +1847,7 @@ func (sa *Application) RemoveAllAllocations() []*Allocation {
 		sa.appEvents.sendRemoveAllocationEvent(alloc, si.TerminationType_STOPPED_BY_RM)
 	}
 
-	if resources.IsZero(sa.pending) {
+	if ugm.GetUserManager().GetUserTracker(sa.user.User) != nil && resources.IsZero(sa.pending) {
```

Comment:
Could you please add comment for this condition? I feel the description of this PR is good example :)